modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -48,14 +50,36 @@ export class Transaction extends BaseTransaction {
   /** @inheritdoc */
   toJson(): TxData {
     const [acc, time] = this.getTxIdParts();
-    return {
+    const result: TxData = {
       id: acc + '@' + time,
       hash: this.getTxHash(), // TODO: Update once hedera-sdk release this functionality BGA-284
       data: toHex(this._hederaTx.bodyBytes),
       fee: new BigNumber(this._txBody.transactionFee!.toString()).toNumber(),
       from: acc,
       startTime: time,
+      validDuration: new BigNumber(this._txBody.transactionValidDuration!.seconds!.toString()).toNumber(),
```

Comment:
there's a lot of number coercement - can we leave a comment here that this is expressed in seconds and add a check just in case this proto doesn't exist?

Reply:
Sure we can just return a string

Reply:
We did this because it's useful for decoding a transaction. Like we do on transactionBuilder.ts line 91. If we don't do that conversion here we need to add it on that other line.